### ISSUE TITLE
SelectorParseException when selecting direct children of a disjunction of queries

### ISSUE THREAD

**Issue by dimo414** [jhy/jsoup#729]
You can limit a selection to children of the current element with `> E` (notice nothing precedes the `>`), however combining this with the `,` operator raises an uninformative parse exception.

If query `A` and `B` succeed then query `A, B` should succeed, however if `A` and `B` are of the form `> E` this fail.

Example test:

``` java
@Test
public void multipleDirect() {
  String html = "<span class=\"foo\">Foo</span><span class=\"bar\">Bar</span>";
  Document doc = Jsoup.parseBodyFragment(html);
  Element body = doc.body();

  assertThat(body.select(".foo, .bar")).hasSize(2);
  assertThat(body.select("> .foo")).hasSize(1);
  assertThat(body.select("> .foo")).hasSize(1);
  assertThat(body.select("body > .foo, body > .bar")).hasSize(2);
  assertThat(body.select("> .foo, > .bar")).hasSize(2); // this line fails
}
```

Stack trace:

```
org.jsoup.select.Selector$SelectorParseException: Could not parse query '': unexpected token at ''
    at org.jsoup.select.QueryParser.findElements(QueryParser.java:195)
    at org.jsoup.select.QueryParser.parse(QueryParser.java:52)
    at org.jsoup.select.QueryParser.parse(QueryParser.java:38)
    at org.jsoup.select.QueryParser.combinator(QueryParser.java:80)
    at org.jsoup.select.QueryParser.parse(QueryParser.java:60)
    at org.jsoup.select.QueryParser.parse(QueryParser.java:38)
    at org.jsoup.select.Selector.<init>(Selector.java:80)
    at org.jsoup.select.Selector.select(Selector.java:93)
    at org.jsoup.nodes.Element.select(Element.java:252)
    at JSoupTest.multipleDirect(JSoupTest.java:60)
    ...
```

**Comment by JenniferJohnson89** [comment]
I noticed that there is a similar problem at https://github.com/jhy/jsoup/issues/1274 and https://github.com/mrozekma/NoiseBot/issues/72. Perhaps we can refer to them to find more context about the bug.

After studying those two issues, I guess the parser was misled by the blank between `> .foo,` and `> .bar` in the last line of your test example and didn't behave normally. The author fixed https://github.com/jhy/jsoup/issues/1274 by adding a trimming step to the parser input, which obviously cannot deal with the blank inside an input. Maybe try eliminating the blank?

Also, I tried to reproduce your bug with the latest version of Jsoup using your example test. However, it seems to throw a new exception: 

`Running org.jsoup.select.SelectorTest
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.069 sec <<< FAILURE!
multipleDirect(org.jsoup.select.SelectorTest)  Time elapsed: 0.037 sec  <<< ERROR!
org.jsoup.select.Selector$SelectorParseException: String must not be empty
        at org.jsoup.select.QueryParser.parse(QueryParser.java:46)
        at org.jsoup.select.QueryParser.combinator(QueryParser.java:89)
        at org.jsoup.select.QueryParser.parse(QueryParser.java:69)
        at org.jsoup.select.QueryParser.parse(QueryParser.java:44)
        at org.jsoup.select.Selector.select(Selector.java:92)
        at org.jsoup.nodes.Element.select(Element.java:373)
        at org.jsoup.select.SelectorTest.multipleDirect(SelectorTest.java:947)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.jsoup.MultiLocaleRule$1.evaluate(MultiLocaleRule.java:23)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:242)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:137)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
        at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
        at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
`

The error info is same as what appears in the jsoup webpage.

**Comment by jhy** [comment]
Thanks - this was fixed in 4ac8a93b0a85f666941ef0198556d1e2395991eb
